### ISSUE TITLE
monetizationstop has same requestId as monetizationpending on payment pointer change

### ISSUE THREAD

**Issue by dacioromero** [coilhq/web-monetization-projects#127]
Changing a payment pointer causes in a `monetizationstop` and `monetizationpending` events to be dispatched where the `monetizationstop` event's `requestId` is the same as `monetizationpending` event's `requestId` rather than the the previous corresponding Monetization events.

This makes the timeline harder to track.

You can observe the behavior here: https://dacioromero.github.io/wm-events/

FYI: the page uses my pointers.

**Comment by sublimator** [comment]
Dang, yeah, there's a race I guess. 
Thanks for the headsup

**Comment by sublimator** [comment]
If you added a small delay after removing one meta then adding another I believe it would work properly, which would show the race I'm talking about. 

Actually, this [re]opens some questions about multiple active streams at once. Theoretically you could initiate a wind down of one stream, start another one immediately (emitting a monetizationpending event), then get a report of a successful packet from the first stream which would usually imply a monetizationprogress event. 

In other words stream events from different monetization requests happening concurrently. 

Currently the implementation just does a hard shutdown of a stream and removes all event listeners so some packets will go unreported, which is less than ideal.
